appservice/src/IAppSettingsClient.ts

```diff
@@ -0,0 +1,15 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { StringDictionary } from 'azure-arm-website/lib/models';
+
+export interface IAppSettingsClient {
```

Comment:
So your goal with this PR is to create a place for code that can and can't be shared. However, it feels like you ended up with _two_ places for code that can't be shared and we only need one. In other words, I don't think you need to create `IAppSettingsClient` _and_ split them out into base classes. Of the two options, I prefer `IAppSettingsClient` because I could see us using that interface for non-tree scenarios as well